types.go

```diff
@@ -19,17 +19,47 @@ type Vertex struct {
 	Value VertexValue `json:"@value"`
 }
 
+func (v1 Vertex) Equals(v2 Vertex) bool {
+	return v1.Type == v2.Type && v1.Value.Equals(v2.Value)
+}
+
 type VertexValue struct {
 	ID         string                      `json:"id"`
 	Label      string                      `json:"label"`
 	Properties map[string][]VertexProperty `json:"properties"`
 }
 
+func (v1 VertexValue) Equals(v2 VertexValue) bool {
+	if v1.ID != v2.ID || v1.Label != v2.Label {
```

Comment:
stylistically seems a bit strange to have these or'd together and then a separate `if` for the length check.  I would say either make them all separate or `||` them all together, but don't mix.  

I think separate if's is easier to read, and is what I'm used to seeing from a java style `equals` method